### PR TITLE
Fixes #404 Color input

### LINKED ISSUE #404: CSS color names appear as black in ColorWidget
When setting a color property to a CSS color like 'white' or 'orange', the color widget displays it as black. The widget handles converting from the short hand hex variant `#333`, but does not handle color names, or HSL values.

I can write a quick patch for this, but wondering what's more desirable -- using a lib like [color-string](https://www.npmjs.com/package/color-string) (+14kb unminified, +500b minified), or putting the supplied color value into a div and using `window.getComputedStyle` and converting by hand the supplied HSL values. Both methods should handle any color format.

### PR DESCRIPTION
**Pull request by fernandojsg**
- Added input box
- Added support for CSS color format: rgb, hsl, hex3, hex6, color names.

/cc @dmarcos @jsantell

### CODE REVIEW

File: src/components/widgets/ColorWidget.js [hunk 1]
```diff
@@ -15,34 +15,77 @@ export default class ColorWidget extends React.Component {
...
+
   onChange = e => {
-    var value = e.target.value;
-    this.setState({value: value});
-    if (this.props.onChange) {
-      this.props.onChange(this.props.name, value);
+    this.setValue(e.target.value);
+  }
+
+  onBlur = e => {
```
Comment (jsantell) [inline src/components/widgets/ColorWidget.js]: `onBlur` is not used

File: src/components/widgets/ColorWidget.js [hunk 2]
```diff
@@ -15,34 +15,77 @@ export default class ColorWidget extends React.Component {
...
+          value={this.state.value}
+          onKeyUp={this.onKeyUp}
+          onChange={this.onChangeText}
+        />
+      </span>
     );
   }
 }
 
 function correctColorValue (value) {
```
Comment (jsantell) [inline src/components/widgets/ColorWidget.js]: `correctColorValue` is no longer used

**Review by jsantell — COMMENTED** [review]
Just pulled down the changes and they work great!

File: src/components/widgets/ColorWidget.js (hunk 1 above)
Comment (fernandojsg) [reply src/components/widgets/ColorWidget.js]: Fixed

File: src/components/widgets/ColorWidget.js (hunk 2 above)
Comment (fernandojsg) [reply src/components/widgets/ColorWidget.js]: Fixed! thanks

**Pull request merged**